http_server/stream_demux_test.py

```diff
@@ -0,0 +1,34 @@
+from amaranth.sim import Simulator

```

Comment:
For "multiple test cases", see also [packet_fixtures_test](https://github.com/cceckman/http-accel/blob/83fa77933c1d8f9d86b6e702f15c2fe4994de193/almost_tcp/packet_fixtures_test.py#L1), which puts more stuff (like DUT) into the test case, and does less in globals. Have to do a little manipulation to uniquify the VCD, but means the tests are really independent.

(No change required, just sharing while I actually prep that for upstream...)